src/NuGet.Core/NuGet.Credentials/CredentialService.cs

```diff
@@ -90,7 +90,7 @@ public CredentialService(AsyncLazy<IEnumerable<ICredentialProvider>> providers,
                 cancellationToken.ThrowIfCancellationRequested();
 
                 var retryKey = RetryCacheKey(uri, type, provider);
-                var isRetry = _retryCache.ContainsKey(retryKey);
+                var isRetry = (type == CredentialRequestType.Forbidden) && _retryCache.ContainsKey(retryKey);
```

Comment:
Could you help me understand why `Forbidden` and not `Unauthorized`?

Forbidden suggests the username&password or PAT is valid, so no amounts of retries are going to succeed until permissions are fixed server side. The edge case is when the customer has two or more accounts, and they entered the wrong one.

Unauthorized suggests the username& password or PAT is invalid, so valid credentials need to be supplied in the retry.

Honestly, my intuition is that both attempts are a kind of retry, but only unauthorized responses make sense to retry. Why shouldn't incorrect credentials being supplied be treated as a retry?

My understanding of the problem, from reading the link issues, is that NuGet doesn't attempt to pass the credentials on the first HTTP request (same as how web browsers (usually?) don't, but NuGet incorrectly detects the first attempt at sending credentials as a retry. My gut feeling is that we need a more sophisticated fix, rather than treating all unauthorized responses as not-retries.

Reply:
> My understanding of the problem, from reading the link issues, is that NuGet doesn't attempt to pass the credentials on the first HTTP request (same as how web browsers (usually?) don't, but NuGet incorrectly detects the first attempt at sending credentials as a retry.

The credential service should only come in play with a challenge, so I don't imagine we have an unneeded isRetry.